src/main/plugins/org.dita.html5/xsl/topic.xsl

```diff
@@ -289,6 +284,33 @@ See the accompanying LICENSE file for applicable license.
   <xsl:template match="*[contains(@class, ' topic/titlealts ')]"/>
   
   
+  <!-- =========== HEADING LEVELS =========== -->
+
+  <!-- returns heading level (1 through 6) appropriate at the specified element -->
+  <xsl:function name="dita2html:get-heading-level" as="xs:integer">
+    <xsl:param name="element" as="element()"/>
+    <xsl:sequence select="min((6, count($element/ancestor-or-self::*[dita2html:is-heading-level(.)])))"/>
+  </xsl:function>
+
+  <!-- returns true() for elements that count as a heading level -->
+  <!-- (this is an accessor function to the moded templates below) -->
+  <xsl:function name="dita2html:is-heading-level" as="xs:boolean?">
+    <xsl:param name="element" as="element()"/>
+    <xsl:apply-templates select="$element" mode="dita2html:is-heading-level"/>
+  </xsl:function>
+  <xsl:mode name="dita2html:is-heading-level" on-no-match="deep-skip"/>
```

Comment:
`<xsl:mode>` is XSLT 3.0 and this stylesheet is 2.0

Reply:
I mentioned in my earlier comment that I was surprised that XSLT3 stuff worked in an XSLT2 template, and I was wondering if you had seen this before. I will revert the constructs to XSLT2. Or maybe I can convince you to change these templates to XSLT3 as a pilot project.  :)